engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnJobBehavior.java

```diff
@@ -121,11 +133,17 @@ private void writeJobCreatedEvent(
     stateWriter.appendFollowUpEvent(jobKey, JobIntent.CREATED, jobRecord);
   }
 
-  private DirectBuffer encodeHeaders(final Map<String, String> taskHeaders, final String assignee) {
+  private DirectBuffer encodeHeaders(
+      final Map<String, String> taskHeaders, final JobProperties props) {
     final var headers = new HashMap<>(taskHeaders);
+    final String assignee = props.getAssignee();
+    final String candidateGroups = props.getCandidateGroups();
     if (assignee != null) {
       headers.put(Protocol.USER_TASK_ASSIGNEE_HEADER_NAME, assignee);
     }
+    if (candidateGroups != null) {
```

Comment:
:thought_balloon: So here we get into the territory I mentioned yesterday - what about variable overlap? Now it is implemented that the explicit fields overwrite the general headers. 

It seems that this is consistent with the way `Protocol.USER_TASK_FORM_KEY_HEADER_NAME` is applied.

However, from a users perspective I would argue it would be more flexible to do it the other way around.

In the end, it seems like you pass a string over. And you take great care that it is a list of string and such. But neither Zeebe nor tasklist really cares about the content of the string, afaik.

So if the user could overwrite the Zeebe default by defining a different header, then the user could pass an arbitrary String. Could be a JSON object, could be an XML file. Could be whatever. 

On the other hand, it needs to be something static which already limits its usefulness.

Anyway, just wanted to share my thoughts on the matter.

Reply:
Thanks for raising this.

>neither Zeebe nor tasklist really cares about the content of the string, afaik.

Tasklist and Optimize may want to parse this in the future.

> if the user could overwrite the Zeebe default by defining a different header, then the user could pass an arbitrary String.

Yeah we should not allow that. It could break things in third party tools. We have a requirement stating:
>custom header with key io.camunda.zeebe:candidateGroups. Its type is a string representation of list of string

🤔 However, we could argue that a user could define a task header `io.camunda.zeebe:candidateGroups` and not use the candidateGroups attribute of AssignmentDefinition which can then break things this way. I guess we should not allow users to set this task header directly. I'll open an issue to continue this topic.

- [ ] open an issue to reject user task with candidategroups task header